### ISSUE TITLE
Feature request: use idiomatic gRPC endpoint URLs

### ISSUE THREAD

**Issue by sander** [higherkindness/mu#599]
Currently, when defining:

```scala
package com.example.ns {
  @outputPackage("com.example.grpcns")
  object protocol {
    // ...
    @service(Protobuf) {
    trait MyService[F[_]] {
      def myMethod(request: MyRequest): F[MyResponse]
    }
  }
}
```

the `idlGen` command produces idiomatic gRPC code:

```proto
// ...
package com.example.grpcns;
// ...
service MyService {
  rpc MyMethod (MyRequest) returns (MyResponse);
}
```

When using `protoc` to generate client code for this, the client calls `com.example.grpcns.MyService/MyMethod` as expected:

* with namespace
* with UpperCamelCase method name

But the Mu server will only accept `MyService/myMethod` calls:

* without a namespace
* with lowerCamelCase method name

This seems to be non-idiomatic gRPC usage and breaks compatibility with IDL-generated clients.

Suggestions for the served gRPC endpoint URLs:

1. Use the @outputPackage annotation to define the namespace.
2. Capitalize the first letter of the method name.

Example project:

- https://github.com/sander/rpctest/tree/f45616a720091d88c06d0a5be00f5a2f4306f7aa doesn’t work, since the Go client tried to reach a prefixed and UpperCamelCase method
- https://github.com/sander/rpctest/tree/e018d5e5d1812b107d976aa7bbfb6617c223c761 does work, but now there is no namespace support and I needed to write UpperCamelCase in Scala which seems un-idiomatic

**Comment by Chicker** [comment]
I faced the same problem during implementing grpc health checks (see https://github.com/grpc/grpc/blob/master/doc/health-checking.md) for our service deployed in Kubernetes. According to the doc above we need to specify concrete namespace `grpc.health.v1` of our implementation of the health check service. But, I don’t know how to do it. I tried to do something like this:

```
package grpc.health.v1

import higherkindness.mu.rpc.protocol._

@service(Protobuf)
trait Health[F[_]] {
  def Check(request: ReqHealthCheck): F[RespHealthCheck]
}
```

But it didn’t help me. I get the same error: `grpc-message: Method not found: grpc.health.v1.Health/Check`

@sander  Did you solve the problem of specifying namespace somehow?

**Comment by sander** [comment]
@Chicker not yet. @fedefernandez said in Gitter chat that he’d try to prioritize this issue.

Haven’t tried, but maybe this works as a workaround?

```scala
trait `grpc.health.v1.Health`[F[_]] {
```

**Comment by Chicker** [comment]
> Haven’t tried, but maybe this works as a workaround?
> 
> ```scala
> trait `grpc.health.v1.Health`[F[_]] {
> ```

@sander Thanks for the good suggestion! But it doesn't work, since  the dot `.` is replaced to the interpolation of the `u+002e` Unicode character. As a result, we get something like this `grpc$u002Ehealth$u002Ev1$u002EHealth` as our serviceName.

**Comment by fedefernandez** [comment]
Thanks for the submission @sander. I'm currently working in this, I'll test it with your sample project.

**Comment by fedefernandez** [comment]
@sander we have added a couple of parameters to the `@service` annotation for controlling the service prefix and method name case. I've created a PR in your demo project to show how to configure.

We're planning to add a new setting key to the idl plugin for controlling that when converting from `avdl` or `proto` to `scala` and update the docs. Once these two tasks are done, we'll release a new version.

**Comment by sander** [comment]
Thanks @fedefernandez! What will the new setting key be for exactly?

**Comment by fedefernandez** [comment]
I'm planning to add a flag that when active, it'll take the namespace defined in the avdl file and put it in the annotation namespace. It'll also set the method name style to capitalize. Still need a name though, ideas are welcomed 😉

**Comment by sander** [comment]
@fedefernandez Is this still blocked on https://github.com/higherkindness/skeuomorph/issues/85 ? Is there a snapshot version that can be used already?

**Comment by fedefernandez** [comment]
@sander sadly is still blocked. Couldn't find any spare time for addressing that.

**Comment by fedefernandez** [comment]
@sander thanks again for your feedback. This issue has been accomplished in #622 and released with the `0.18.4` version (it'll reach maven central soon)

Now you can set the `idlGenIdiomaticEndpoints` setting key to `true` for generating scala service definitions that will append the namespace to the operations and their names will be capitalized.
